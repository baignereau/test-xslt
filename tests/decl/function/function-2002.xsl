<?xml version="1.0"?> 

<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0">

  <!-- test that all new functions listed in the XSLT specification are not implemented (or claim to be available!) in 2.0 -->
  <!-- This test uses xslt.xml as input. This should be updated from time to time
     with the latest version (modified by removing the DTD reference) -->


  <xsl:template match="/">
    <out>
      <xsl:apply-templates select=".//proto"/>
    </out>
  </xsl:template>

  <xsl:template match="proto">
    <xsl:variable name="lexname" select="@name"/>
    <f name="{$lexname}" arity="{count(arg)}" test="{function-available($lexname, count(arg))}"/>
  </xsl:template>
  
</xsl:stylesheet>
