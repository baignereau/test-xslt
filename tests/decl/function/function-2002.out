<?xml version="1.0" encoding="UTF-8"?><out><f name="current-group" arity="0" test="true"/><f name="current-grouping-key" arity="0" test="true"/><f name="current-merge-inputs" arity="1" test="false"/><f name="regex-group" arity="1" test="true"/><f name="copy-of" arity="0" test="false"/><f name="copy-of" arity="1" test="false"/><f name="snapshot" arity="0" test="false"/><f name="snapshot" arity="1" test="false"/><f name="outermost" arity="1" test="false"/><f name="innermost" arity="1" test="false"/><f name="has-children" arity="0" test="false"/><f name="document" arity="1" test="true"/><f name="document" arity="2" test="true"/><f name="uri-collection" arity="0" test="false"/><f name="uri-collection" arity="1" test="false"/><f name="unparsed-text" arity="1" test="true"/><f name="unparsed-text" arity="2" test="true"/><f name="unparsed-text-lines" arity="1" test="false"/><f name="unparsed-text-lines" arity="2" test="false"/><f name="unparsed-text-available" arity="1" test="true"/><f name="unparsed-text-available" arity="2" test="true"/><f name="key" arity="2" test="true"/><f name="key" arity="3" test="true"/><f name="format-number" arity="2" test="true"/><f name="format-number" arity="3" test="true"/><f name="current" arity="0" test="true"/><f name="unparsed-entity-uri" arity="1" test="true"/><f name="unparsed-entity-public-id" arity="1" test="true"/><f name="system-property" arity="1" test="true"/><f name="format-dateTime" arity="5" test="true"/><f name="format-dateTime" arity="2" test="true"/><f name="format-date" arity="5" test="true"/><f name="format-date" arity="2" test="true"/><f name="format-time" arity="5" test="true"/><f name="format-time" arity="2" test="true"/><f name="function-available" arity="1" test="true"/><f name="function-available" arity="2" test="true"/><f name="type-available" arity="1" test="true"/><f name="element-available" arity="1" test="true"/></out>